STCommon/src/commonMain/kotlin/com/infomaniak/multiplatform_swisstransfer/common/interfaces/ui/FileUi.kt

```diff
@@ -20,23 +20,49 @@ package com.infomaniak.multiplatform_swisstransfer.common.interfaces.ui
 import com.infomaniak.multiplatform_swisstransfer.common.interfaces.transfers.File
 import com.infomaniak.multiplatform_swisstransfer.common.interfaces.upload.RemoteUploadFile
 import com.infomaniak.multiplatform_swisstransfer.common.interfaces.upload.UploadFileSession
+import kotlin.uuid.ExperimentalUuidApi
+import kotlin.uuid.Uuid
 
 data class FileUi(
     val uid: String,
     val fileName: String,
+    val isFolder: Boolean? = false,
     val fileSize: Long,
     val mimeType: String?,
     val localPath: String?,
+    var parent: File? = null,
+    var children: MutableList<File> = mutableListOf(),
 ) {
 
     constructor(file: File) : this(
         uid = file.uuid,
         fileName = file.fileName,
+        isFolder = file.isFolder,
         fileSize = file.receivedSizeInBytes,
         mimeType = file.mimeType,
         localPath = null,
     )
 
+    // Init for folder
+    @OptIn(ExperimentalUuidApi::class)
+    constructor(folderName: String) : this(
```

Comment:
The FileUI is only a wrapper for FileDB used for app ui, then we need to only create a FileUI from a FileDB